src/components/AddItemForm.tsx

```diff
@@ -1,9 +1,47 @@
 import React from 'react';
 
-function AddItemForm() {
+type TaskListItem = {
+  task: string;
+  completed: boolean;
+  id: string;
+};
+interface TaskListItemMap {
+  [value: string | number]: TaskListItem;
+}
+interface AddItemFormProps {
+  userInput: string;
+  setUserInput: (value: string) => void;
+  taskList: TaskListItemMap;
+  setTaskList: (value: any) => void;
+}
+
+function AddItemForm({
+  userInput,
+  setUserInput,
+  taskList,
+  setTaskList,
+}: AddItemFormProps) {
+  function getNewTaskList() {
+    const uniqueTaskID = String(new Date().valueOf());
+    const typedTask = userInput;
+    const myReturn: any = { ...taskList };
+    myReturn[uniqueTaskID] = {
+      task: typedTask,
+      completed: false,
+      id: uniqueTaskID,
+    };
+    return myReturn;
+  }
   const domInputId = 'new-todo-input';
+  function handleAddTask(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (userInput) {
+      setTaskList(getNewTaskList());
+    }
+  }
```

Comment:
Como mecionei no código do App, utilizando a função addTask:
```suggestion
  const domInputId = 'new-todo-input';
  function handleAddTask(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();

    if (userInput) {
      const uniqueTaskID = String(new Date().valueOf());
      const newTask = {
        task: userInput,
        completed: false,
        id: uniqueTaskID,
      };
      addTask(newTask, uniqueTaskID);
    }
  }
```

Assim, será necessário apenas passar o update para a função, e não o estado como um todo. ^^

Considere que isto se relaciona também com o formato do `taskList`. Perceba que tanto no meu código quanto no seu estamos replicando a chave `uniqueTaskID` em dois níveis do nosso objeto, precisamos mesmo deste dado salvo 2x? Daqui nasce a sugestão de mudança para uma Task[] ou seja:
`[{id: uniqueTaskID, completed: false, task: typedTask }]`